### ISSUE TITLE
focal loss Mean Issue

### ISSUE THREAD

**Issue by jaideep11061982** [xingyizhou/CenterNet#546]
Hi xing
in your focal loss i see you are divding all the no of positives that there are in batch.
loss = loss - (pos_loss + neg_loss) / num_pos
Rather it should be  loss/image / num of positives  in image mask.
 And also  we should leave loss at the image mask level  and calculate mean at batch level . 
Say There are 3 images in batch

so I1 total loss is x  
   I1 total loss is y
I3 total loss is z
Currently we intent to x/total pos   in x and like wise

Expected should be
(x+y+z)/3

**Comment by xingyizhou** [comment]
Thanks for pointing out the issue. The focal loss is from the CornerNet paper. However I do have tried the version your are proposing. The resulting loss is in a larger scale. Without other hyper-parameter change, it does not converge.

**Comment by jaideep11061982** [comment]
Hi Xing
Thanks for spending time in going through.
I experienced troubled with this version while using it for kaggles Car 6d
pose estimation competition where labels dim were of size H,W <input image
size and  will have non zero pixel at the position where there are cars in
image  .When we do mean in the current way model is unable to learn any
thing but when its removed that is num pos in divide, things work and
converge as expected.
I still dont have reasoning why is it so  Even  pytorchs standard BCE fails
in this case but same works very well in other segmentation problems.

**Comment by xingyizhou** [comment]
I didn't understand your reply, but it seems the issue is resolved.
